packages/components/src/navigation/NavTab.tsx

```diff
@@ -18,6 +18,7 @@ interface NavTabProps {
   index: number;
   isDraggable: boolean;
   contextActions?: ResolvableContextAction | ResolvableContextAction[];
+  renderTabSlot?: (tab: NavTabItem) => React.ReactNode;
```

Comment:
Might as well add a jsdoc comment above this prop so it shows up with intellisense.

Also, thoughts on `renderAfterContent` or something a little more descriptive?